eplus.py

```diff
@@ -37,6 +44,48 @@ def _get_eplus_data(session, eplus_url):
     return result
 
 
+class EplusSessionUpdater(Thread):
+    def __init__(self, session, eplus_url):
+        self._eplus_url = eplus_url
+        self._session = session
+        self._closed = Event()
+
+        super().__init__(name='EplusSessionUpdater', daemon=True)
+
+    def close(self):
+        log.info('[EplusSessionUpdater] Closing...')
+        self._closed.set()
+
+    def run(self):
+        while True:
+            if self._closed.is_set():
+                return
+
+            # Create a new session, and send a request to Eplus url to obtain the cookies4
+            log.debug('[EplusSessionUpdater] Refreshing cookies...')
+            fresh_response = self._session.http.get(self._eplus_url, headers={
+                'Cookie': ''
+            })
+
+            # Update the session with the new cookies
+            self._session.http.cookies.clear()
+            self._session.http.cookies.update(fresh_response.cookies)
+            log.debug(f'[EplusSessionUpdater] Successfully updated cookies: <{repr(fresh_response.cookies)}>')
+
+            self.wait_for_about_half_hour()
+
+    def wait_for_about_half_hour(self):
+        """
+        Cookie of the Eplus expires after about 1 hour.
+        To keep the cookie fresh, we'll have to refresh the cookies every half hour.
+
+        e.g. To avoid multiple people visiting Eplus at the same time, a random interval is used.
+        """
+        wait_sec = random.randint(20 * 60, 40 * 60)
```

Comment:
I don't think this randomization is necessary, using a regular fixed timeout should be fine. There aren't many people using this plugin, and even if there were a lot of users, we still look basically the same as any other browser refreshing on a regular interval.

But I think checking the cookie expiration time and then waiting until a few minutes before it expires would be even better than a fixed timeout.